src/Clide/Commands/CommandManager.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.ComponentModel.Design;
+using System.Linq;
+
+namespace Clide.Commands
+{
+	[Export(typeof(ICommandManager))]
+	class CommandManager : ICommandManager
+	{
+		readonly Lazy<IMenuCommandService> menuCommandService;
+		readonly Lazy<IServiceLocator> serviceLocator;
+
+		[ImportingConstructor]
+		public CommandManager(
+			[Import(ContractNames.Interop.IMenuCommandService)] Lazy<IMenuCommandService> menuCommandService,
+			Lazy<IServiceLocator> serviceLocator)
+		{
+			this.menuCommandService = menuCommandService;
+			this.serviceLocator = serviceLocator;
+		}
+
+		public void RegisterCommands(IServiceProvider package)
+		{
+			var packageGuid = package.GetType().GUID;
+
+			var commands = serviceLocator
+				.Value
+				.GetExports<ICommandExtension, ICommandMetadata>()
+				.Where(x => new Guid(x.Metadata.PackageId) == packageGuid)
```

Comment:
any chance we can avoid newing up all those `Guid`s on registration? If there are many, this might impact perf. Can we just assume the user will use the same string constant for both and therefore a simple string comparison will work? maybe make it `OrdinalIgnoreCase`?